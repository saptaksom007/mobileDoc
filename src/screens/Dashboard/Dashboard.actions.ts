import { Tokens } from 'api/types'
import { RouteAction } from 'navigation/SagaNavigation'

export const ActionTypes = {
  ENABLE_NOTIFICATION_REQUESTED: 'ENABLE_NOTIFICATION_REQUESTED',
  ENABLE_NOTIFICATION_SUCCEEDED: 'ENABLE_NOTIFICATION_SUCCEEDED',
  ENABLE_NOTIFICATION_FAILED: 'ENABLE_NOTIFICATION_FAILED',
  LOGOUT_REQUESTED: 'LOGOUT_REQUESTED',
  LOGOUT_SUCCEEDED: 'LOGOUT_SUCCEEDED',
  LOGOUT_FAILED: 'LOGOUT_FAILED',
  REFRESH_TOKEN_REQUESTED: 'REFRESH_TOKEN_REQUESTED',
  REFRESH_TOKEN_SUCCEEDED: 'REFRESH_TOKEN_SUCCEEDED',
  REFRESH_TOKEN_FAILED: 'REFRESH_TOKEN_FAILED',
  REFRESH_TOKEN_CANCELLED: 'REFRESH_TOKEN_CANCELLED',
  APP_STATE_CHANGE: 'APP_STATE_CHANGE',
  CONNECTIVITY_CHANGE: 'CONNECTIVITY_CHANGE',
  GOTO_LOCKED_DASHBOARD_REQUESTED: 'GOTO_LOCKED_DASHBOARD_REQUESTED',
  GOTO_LOCKED_DASHBOARD_SUCCEEDED: 'GOTO_LOCKED_DASHBOARD_SUCCEEDED',
  GOTO_LOCKED_DASHBOARD_FAILED: 'GOTO_LOCKED_DASHBOARD_FAILED',
  LOGIN_WITH_NEXT_ROUTE_REQUESTED: 'LOGIN_WITH_NEXT_ROUTE_REQUESTED',
  LOGIN_WITH_NEXT_ROUTE_SUCCEEDED: 'LOGIN_WITH_NEXT_ROUTE_SUCCEEDED',
  LOGIN_WITH_NEXT_ROUTE_FAILED: 'LOGIN_WITH_NEXT_ROUTE_FAILED',
  NEXT_ROUTE_SUCCEEDED: 'NEXT_ROUTE_SUCCEEDED',
  NEXT_ROUTE_FAILED: 'NEXT_ROUTE_FAILED',
  SHOW_VERIFY_NOTIFS_REQUESTED: 'SHOW_VERIFY_NOTIFS_REQUESTED',
  HIDE_VERIFY_NOTIFS_REQUESTED: 'HIDE_VERIFY_NOTIFS_REQUESTED',
  SHOW_TERMS: 'SHOW_TERMS',
  HIDE_TERMS: 'HIDE_TERMS',
  NEW_APP_VERSION_REQUESTED: 'NEW_APP_VERSION_REQUESTED',
  NEW_APP_VERSION_SUCCEEDED: 'NEW_APP_VERSION_SUCCEEDED',
  NEW_APP_VERSION_FAILED: 'NEW_APP_VERSION_FAILED',
  REFRESH_DASHBOARD: 'REFRESH_DASHBOARD',
  GET_LOGGED_IN_USER_DETAILS_REQUESTED: 'GET_LOGGED_IN_USER_DETAILS_REQUESTED',
  GET_LOGGED_IN_USER_DETAILS_SUCCEEDED: 'GET_LOGGED_IN_USER_DETAILS_SUCCEEDED',
  GET_LOGGED_IN_USER_DETAILS_FAILED: 'GET_LOGGED_IN_USER_DETAILS_FAILED',
}

export default class Actions {
  static logOutRequested(clearStorage: boolean = false) {
    return { type: ActionTypes.LOGOUT_REQUESTED, clearStorage }
  }
  static logOutSucceeded() {
    return { type: ActionTypes.LOGOUT_SUCCEEDED }
  }
  static logOutFailed(error: Error) {
    return {
      type: ActionTypes.LOGOUT_FAILED,
      message: error.message,
      error,
    }
  }
  static enableNotificationsRequested() {
    return { type: ActionTypes.ENABLE_NOTIFICATION_REQUESTED }
  }
  static enableNotificationsSucceeded() {
    return { type: ActionTypes.ENABLE_NOTIFICATION_SUCCEEDED }
  }
  static enableNotificationsFailed(error: Error) {
    return {
      type: ActionTypes.ENABLE_NOTIFICATION_FAILED,
      message: error.message,
      error,
    }
  }

  static refreshTokenSucceeded(result: Tokens) {
    return { type: ActionTypes.REFRESH_TOKEN_SUCCEEDED, result }
  }
  static refreshTokenCancelled() {
    return { type: ActionTypes.REFRESH_TOKEN_CANCELLED }
  }
  static refreshTokenFailed(error: Error) {
    return {
      type: ActionTypes.REFRESH_TOKEN_FAILED,
      message: error.message,
      error,
    }
  }
  static gotoLockedDashboardRequested() {
    return { type: ActionTypes.GOTO_LOCKED_DASHBOARD_REQUESTED }
  }
  static gotoLockedDashboardSucceeded() {
    return { type: ActionTypes.GOTO_LOCKED_DASHBOARD_SUCCEEDED }
  }
  static gotoLockedDashboardFailed(error: Error) {
    return {
      type: ActionTypes.GOTO_LOCKED_DASHBOARD_FAILED,
      message: error.message,
      error,
    }
  }
  static loginWithNextRouteRequested(nextRoute: RouteAction) {
    return {
      type: ActionTypes.LOGIN_WITH_NEXT_ROUTE_REQUESTED,
      payload: { nextRoute },
    }
  }
  static loginWithNextRouteSucceeded() {
    return { type: ActionTypes.LOGIN_WITH_NEXT_ROUTE_SUCCEEDED }
  }
  static loginWithNextRouteFailed(error: Error) {
    return {
      type: ActionTypes.LOGIN_WITH_NEXT_ROUTE_FAILED,
      message: error.message,
      error,
    }
  }
  static nextRouteSucceeded() {
    return { type: ActionTypes.NEXT_ROUTE_SUCCEEDED }
  }
  static nextRouteFailed(error: Error) {
    return {
      type: ActionTypes.NEXT_ROUTE_FAILED,
      message: error.message,
      error,
    }
  }
  static showVerifyNotificationRequested() {
    return { type: ActionTypes.SHOW_VERIFY_NOTIFS_REQUESTED }
  }
  static hideVerifyNotificationRequested() {
    return { type: ActionTypes.HIDE_VERIFY_NOTIFS_REQUESTED }
  }
  static showTerms() {
    return { type: ActionTypes.SHOW_TERMS }
  }
  static hideTerms() {
    return { type: ActionTypes.HIDE_TERMS }
  }

  static newAppVersionRequested() {
    return { type: ActionTypes.NEW_APP_VERSION_REQUESTED }
  }
  static newAppVersionSucceeded(reason?: string) {
    return { type: ActionTypes.NEW_APP_VERSION_SUCCEEDED, payload: reason }
  }
  static newAppVersionFailed(error: Error) {
    return { type: ActionTypes.NEW_APP_VERSION_FAILED, payload: { error } }
  }
  static refreshDashboard() {
    return { type: ActionTypes.REFRESH_DASHBOARD, payload: undefined }
  }

  static loggedInUserDetails() {
    return { type: ActionTypes.GET_LOGGED_IN_USER_DETAILS_REQUESTED }
  }

  static loggedInUserDetailsSucceeded(data?: any) {
    return {
      type: ActionTypes.GET_LOGGED_IN_USER_DETAILS_SUCCEEDED,
      payload: data,
    }
  }
  static loggedInUserDetailsFailed(error: Error) {
    return {
      type: ActionTypes.GET_LOGGED_IN_USER_DETAILS_FAILED,
      payload: { error },
    }
  }
}
