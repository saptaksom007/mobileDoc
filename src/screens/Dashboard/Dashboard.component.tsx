import React, { Component } from 'react'

import { View, ScrollView, InteractionManager, StatusBar } from 'react-native'
import { connect } from 'react-redux'
import { compose as Rcompose, pipe, length, head, propOr } from 'ramda'

import { compose } from 'recompose'

import { Actions as NavAction } from 'navigation/SagaNavigation'
import ConversationsListActions from 'screens/ConversationsList/ConversationsList.actions'

import withoutKeyboard from 'screens/enhancers/withoutKeyboard'

import { getTestId } from 'utilities/environment'

import uncheckedMessages from 'common-docdok/lib/domain/messaging/selectors/uncheckedMessages'
import primaryConSelector from 'common-docdok/lib/domain/messaging/selectors/primaryConSelector'
import { getClinicsFromAvailabilities } from 'common-docdok/lib/domain/healthrelation/selectors/findClinic'
import {
  hasOnlyOneConversation,
  isChat,
} from 'common-docdok/lib/domain/messaging/selectors/findConversation'
import { getNumberOfIncompleteSurvey } from 'common-docdok/lib/domain/survey/selectors/incompleteSurveys'

import AppointmentsListActions from 'screens/AppointmentsList/AppointmentsList.actions'
import Actions from './Dashboard.actions'
import styles from './Dashboard.styles'
import Box from './components/Box'
import { Action } from 'redux'
import { filterOldAppointment, filterMarkAsRead } from 'api/appointments'
import { ConversationDtoType } from 'common-docdok/lib/types'
import { UserOrigins } from 'constants/UserOrigins'
import { Color } from 'constants/Color'
import { Layout } from 'constants/Layout'
import CalendarEventActions from 'screens/CalendarEvent/CalendarEvent.actions'
import { FloatingAction } from 'react-native-floating-action'
import BookAppointmentActions from 'screens/BookAppointment/BookAppointment.actions'
import calendarConstants from 'screens/CalendarEvent/constants/calendarConstants'

const testIdDashboard = getTestId('dashboard')
const testIdConversations = getTestId('dashboard-nav-conversation')
const testIdSurvey = getTestId('dashboard-nav-survey')
const testIdAppointment = getTestId('dashboard-nav-appointment')
const testIdDossier = getTestId('dashboard-nav-dossier')
const testIdAvailability = getTestId('dashboard-nav-availability')

interface Props {
  dispatch(action: Action): void
  incompleteSurveys: number
  unreadMessages: number
  futureAppointmentsCount: number
  primaryConversation: any
  isLoggedIn?: boolean
  onlyOneConversation: boolean
  numberOfPatients: number
  patient: any
  clinics: any[]
  isLoading?: boolean
  rememberedGoToChatAction?: Action
  origin: 'SPITALSTSAG' | undefined
  forceConversationId: number
  primaryConversationId: number
  scheduleAppointment: number
  patients: Array<{ firstName: string; clinicId: number }>
}

export class Dashboard extends Component<Props> {
  componentDidMount() {
    const { dispatch, isLoggedIn, rememberedGoToChatAction } = this.props
    StatusBar.setHidden(false)
    StatusBar.setBarStyle('dark-content')
    if (isLoggedIn) {
      if (rememberedGoToChatAction) {
        dispatch(ConversationsListActions.execRememberedGotoChatRequested())
      }
      InteractionManager.runAfterInteractions(() => {
        // warning this succeeded action
        // with next route feature on login
        dispatch(Actions.enableNotificationsRequested())
        dispatch(AppointmentsListActions.appointmentsListRequested())
        // dispatch(Actions.loggedInUserDetails())
      })
    }
  }

  chatAction = () => {
    const { primaryConversation, onlyOneConversation } = this.props
    if (onlyOneConversation && primaryConversation) {
      return ConversationsListActions.gotoChatRequested(primaryConversation.id)
    }
    return NavAction.push('conversationslist')
  }

  render() {
    const {
      incompleteSurveys,
      unreadMessages,
      futureAppointmentsCount,
      isLoggedIn,
      numberOfPatients,
      patient,
      clinics,
      origin,
      dispatch,
      onlyOneConversation,
      forceConversationId,
      primaryConversationId,
      patients,
      scheduleAppointment,
    } = this.props
    const noHospital = origin === 'SPITALSTSAG'

    const isMigros = false && origin === UserOrigins.MIGROS

    return (
      // <ImageBackground
      //   source={require('assets/images/Background.png')}
      //   style={styles.bg}
      // >
      <>
        <ScrollView
          contentContainerStyle={styles.container}
          testID={testIdDashboard}
          accessibilityLabel={testIdDashboard}
        >
          {/* <View
            style={{
              backgroundColor: '#1a7dae',
              height: Layout.window.height * 0.12,
              justifyContent: 'center',
            }}
          >
            <View
              style={{
                flexDirection: 'row',
                justifyContent: 'flex-start',
                alignItems: 'flex-start',
              }}
            >
              <View
                style={{
                  paddingHorizontal: 15,
                  paddingVertical: 12,
                }}
              >
                <Icon
                  name={'notifications-none'}
                  size={34}
                  color={Color.white}
                />
              </View>
              <View
                style={{
                  justifyContent: 'flex-start',
                  alignItems: 'flex-start',
                  paddingRight: 40,
                  paddingHorizontal: 15,
                  paddingVertical: 7,
                }}
              >
                <Bold style={{ color: Color.white }}>
                  Hi {patientFirstName}
                </Bold>

                <Paragraph style={{ color: Color.white }}>
                  Tap here to see 3 items on your agenda today
                </Paragraph>
              </View>
            </View>
          </View> */}
          <View
            style={{
              flexDirection: 'row',
              justifyContent: 'space-between',
              paddingHorizontal: Layout.window.width * 0.05,
            }}
          >
            <Box
              name={'event'}
              badgeNumber={futureAppointmentsCount}
              nextRoute={() =>
                CalendarEventActions.navigateToCalendarEventList(
                  Boolean(onlyOneConversation || forceConversationId),
                  forceConversationId || primaryConversationId,
                  patients[0].clinicId,
                  scheduleAppointment,
                  futureAppointmentsCount,
                )
              }
              testID={testIdAppointment}
              origin={origin}
              color={'#8950b7'}
            />

            <Box
              name={'question-answer'}
              badgeNumber={unreadMessages}
              testID={testIdConversations}
              locked={!isLoggedIn}
              nextRoute={this.chatAction}
              origin={origin}
              color={'#1a7dae'}
            />
          </View>
          <View
            style={{
              flexDirection: 'row',
              justifyContent: 'space-between',
              paddingHorizontal: Layout.window.width * 0.05,
            }}
          >
            <Box
              name={'stacked-line-chart'}
              locked={!isLoggedIn}
              nextRoute={() => NavAction.push('comingsoon')}
              testID={testIdSurvey}
              origin={origin}
              color={'#fba81f'}
            />
            {/* <Box
              name={'show-chart'}
              badgeNumber={incompleteSurveys}
              locked={!isLoggedIn}
              nextRoute={() =>
                numberOfPatients === 1
                  ? NavAction.push('survey', { patientId: patient.uuid })
                  : NavAction.push('surveybydossier')
              }
              testID={testIdSurvey}
              origin={origin}
              color={'#1a7dae'}
            /> */}
            <Box
              badgeNumber={incompleteSurveys}
              name={'description'}
              locked={!isLoggedIn}
              nextRoute={() =>
                numberOfPatients === 1
                  ? NavAction.push('survey', { patientId: patient.uuid })
                  : NavAction.push('surveybydossier')
              }
              testID={testIdDossier}
              origin={origin}
              color={'#27debe'}
            />
            {/* <Box
            name={'bar-chart'}
            nextRoute={() => NavAction.push('myprogress')}
            testID={testIdAppointment}
            origin={origin}
            color={'#1a7dae'}
          /> */}
          </View>
          <View
            style={{
              flexDirection: 'row',
              justifyContent: 'space-between',
              paddingHorizontal: Layout.window.width * 0.05,
            }}
          >
            {!noHospital && !isMigros ? (
              <>
                <Box
                  name={'escalator-warning'}
                  locked={!isLoggedIn}
                  nextRoute={() => NavAction.push('dossier')}
                  testID={testIdDossier}
                  origin={origin}
                  color={'#2a6df7'}
                />
                <Box
                  name={'officehours'}
                  locked={!isLoggedIn}
                  nextRoute={() =>
                    clinics.length === 1
                      ? NavAction.push('availabilitydetail', { ...clinics[0] })
                      : NavAction.push('availabilitylist')
                  }
                  testID={testIdAvailability}
                  color={'#f37a9e'}
                />
              </>
            ) : null}
          </View>

          {/* {!isSmallScreen() && (
            <View
              style={[
                styles.bottomLinks,
                noHospital && { bottom: 70, justifyContent: 'center' },
              ]}
            >
              <MessageMyDoctor
                containerStyle={styles.messageMyDoctor}
                origin={origin}
              />
              {!noHospital ? <BookAppointmentLink origin={origin} /> : null}
            </View>
          )} */}
        </ScrollView>
        <FloatingAction
          color={Color.blueD}
          actions={calendarConstants.fabActions}
          onPressItem={name => {
            if (name === 'appointments') {
              dispatch(
                BookAppointmentActions.navigateToBookAppointment(
                  forceConversationId || primaryConversationId,
                  patients?.[0]?.clinicId,
                  scheduleAppointment,
                ),
              )
            } else {
              dispatch(
                CalendarEventActions.navigateToCalendarEventLoadList(
                  String(name),
                ),
              )
            }
          }}
        />
      </>
      // </ImageBackground>
    )
  }
}

const getNumberOfFutureAppointments = length

const getFutureAppointmentId = pipe<string, string, number>(
  head,
  propOr(0, 'id'),
)

function select(state: any): any {
  const {
    survey: { mySurveys },
    login: { isLoggedIn },
    appointmentslist,
    messaging,
    profile,
    healthrelation: { patients, availabilities },
    loadingStack: { isLoading },
    conversation: { rememberedGoToChatAction },
    scheduleappointment,
  } = state
  const conversations = Object.values(
    messaging.conversations,
  ) as ConversationDtoType[]
  const unreadMessages = conversations
    .filter(isChat)
    .map((c: any) => c.id)
    .reduce((current, id) => current + uncheckedMessages(messaging, id), 0)

  const currentUser = profile && profile.uid
  const origin = profile && profile.origin

  const patient = patients.filter((p: any) => p.userRef === currentUser)[0]
  const primaryConversationId = patient
    ? primaryConSelector(state, patient.uuid)
    : undefined
  const primaryConversation = primaryConversationId
    ? messaging.conversations[primaryConversationId]
    : undefined

  const incompleteSurveys = getNumberOfIncompleteSurvey(mySurveys || [])

  const futureAppointmentsCount = getNumberOfFutureAppointments(
    // @ts-ignore
    Rcompose(
      filterOldAppointment,
      // @ts-ignore
      filterMarkAsRead,
    )(appointmentslist.rawList || []),
  )

  const numberOfPatients = (patients && patients.length) || 0

  const appointmentId = getFutureAppointmentId(
    // @ts-ignore
    filterOldAppointment(appointmentslist.rawList || []),
  )

  const onlyOneConversation = hasOnlyOneConversation(
    Object.values(messaging.conversations),
  )

  const clinics = getClinicsFromAvailabilities(availabilities)

  return {
    unreadMessages: isLoggedIn ? unreadMessages : 0,
    incompleteSurveys: isLoggedIn ? incompleteSurveys : 0,
    primaryConversation,
    onlyOneConversation,
    futureAppointmentsCount,
    numberOfPatients,
    appointmentId,
    isLoggedIn,
    patient,
    clinics,
    isLoading,
    rememberedGoToChatAction,
    origin,
    patients,
    primaryConversationId,
    scheduleAppointment: scheduleappointment,
  }
}

export default compose<Props, any>(withoutKeyboard, connect(select))(Dashboard)
