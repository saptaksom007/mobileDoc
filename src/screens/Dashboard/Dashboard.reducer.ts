import { pathOr } from 'ramda'
import { userActionTypes } from 'common-docdok/lib/domain/user/actions'

import { ActionTypes } from './Dashboard.actions'

const {
  ENABLE_NOTIFICATION_REQUESTED,
  ENABLE_NOTIFICATION_SUCCEEDED,
  ENABLE_NOTIFICATION_FAILED,
  REFRESH_TOKEN_FAILED,
  LOGOUT_FAILED,
  GOTO_LOCKED_DASHBOARD_SUCCEEDED,
  APP_STATE_CHANGE,
  CONNECTIVITY_CHANGE,
  LOGIN_WITH_NEXT_ROUTE_REQUESTED,
  NEXT_ROUTE_SUCCEEDED,
  NEXT_ROUTE_FAILED,
  SHOW_VERIFY_NOTIFS_REQUESTED,
  HIDE_VERIFY_NOTIFS_REQUESTED,
  SHOW_TERMS,
  HIDE_TERMS,
  GET_LOGGED_IN_USER_DETAILS_REQUESTED,
  GET_LOGGED_IN_USER_DETAILS_SUCCEEDED,
  GET_LOGGED_IN_USER_DETAILS_FAILED,
} = ActionTypes

const {
  UPDATE_TERMS_VERSION_SUCCEEDED,
  TERMS_VERSION_CHANGED,
  TERMS_VERSION_EQUALS,
} = userActionTypes

interface RouteType {
  type: 'NAV_PUSH_REQUESTED'
  route: string
  params?: any
}

interface State {
  error?: string
  limitedAccess: boolean
  currentAppState?: string
  connectivityState?: boolean
  nextRoute?: RouteType
  isVerifNotifDisplay?: boolean
  agreeTerm: boolean
  termsVersion?: string
  loggedInUserDetails?: any
}

interface Action {
  type: keyof typeof ActionTypes
  message?: string
  currentAppState: string
  payload?: any
}

const initialState: State = {
  error: undefined,
  limitedAccess: false,
  currentAppState: undefined,
  connectivityState: true,
  nextRoute: undefined,
  isVerifNotifDisplay: false,
  agreeTerm: false,
  termsVersion: undefined,
  loggedInUserDetails: null,
}

const DEFAULT_TERMS_VERSION = '2018-04-03'

export default function dashboard(state: State = initialState, action: Action) {
  switch (action.type) {
    case ENABLE_NOTIFICATION_REQUESTED: {
      return {
        ...state,
        error: undefined,
      }
    }

    case ENABLE_NOTIFICATION_SUCCEEDED: {
      return {
        ...state,
        error: undefined,
      }
    }

    case ENABLE_NOTIFICATION_FAILED: {
      return {
        ...state,
        error: action.message,
      }
    }

    case REFRESH_TOKEN_FAILED: {
      return { ...state, error: action.message }
    }

    case LOGOUT_FAILED: {
      return { ...state, error: action.message }
    }

    case GOTO_LOCKED_DASHBOARD_SUCCEEDED: {
      return { ...state, limitedAccess: true }
    }

    case APP_STATE_CHANGE: {
      const { currentAppState } = action
      return { ...state, currentAppState }
    }

    case CONNECTIVITY_CHANGE: {
      const connectivityState: boolean = pathOr(
        true,
        ['payload', 'isConnected'],
        action,
      )
      return { ...state, connectivityState }
    }

    case TERMS_VERSION_EQUALS: {
      return { ...state, agreeTerm: true }
    }

    case TERMS_VERSION_CHANGED: {
      const termsVersion: string = pathOr(
        DEFAULT_TERMS_VERSION,
        ['payload', 'termsVersion'],
        action,
      )
      return { ...state, agreeTerm: false, termsVersion }
    }

    case UPDATE_TERMS_VERSION_SUCCEEDED: {
      return { ...state, agreeTerm: true }
    }

    case LOGIN_WITH_NEXT_ROUTE_REQUESTED: {
      const nextRoute: RouteType = pathOr(
        undefined,
        ['payload', 'nextRoute'],
        action,
      )
      return { ...state, nextRoute }
    }

    case NEXT_ROUTE_SUCCEEDED: {
      return { ...state, nextRoute: undefined }
    }

    case NEXT_ROUTE_FAILED: {
      return { ...state, nextRoute: undefined }
    }

    case SHOW_VERIFY_NOTIFS_REQUESTED: {
      return { ...state, isVerifNotifDisplay: true }
    }

    case HIDE_VERIFY_NOTIFS_REQUESTED: {
      return { ...state, isVerifNotifDisplay: false }
    }

    case SHOW_TERMS: {
      return { ...state, showTerms: true }
    }

    case HIDE_TERMS: {
      return { ...state, showTerms: false }
    }

    case GET_LOGGED_IN_USER_DETAILS_REQUESTED: {
      return {
        ...state,
        loggedInUserDetails: null,
        error: undefined,
      }
    }

    case GET_LOGGED_IN_USER_DETAILS_SUCCEEDED: {
      return {
        ...state,
        loggedInUserDetails: action.payload,
        error: undefined,
      }
    }

    case GET_LOGGED_IN_USER_DETAILS_FAILED: {
      return {
        ...state,
        loggedInUserDetails: null,
        error: action.message,
      }
    }

    default:
      return state
  }
}
