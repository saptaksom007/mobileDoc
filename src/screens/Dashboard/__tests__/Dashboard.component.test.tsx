import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { Dashboard } from '../Dashboard.component'

jest.mock('components/Avatar/Avatar.component', () => null)
jest.mock('../components/Box', () => 'Box')
jest.mock('components/MessageMyDoctor/MessageMyDoctor.component')
jest.mock(
  'components/BookAppointmentLink/BookAppointmentLink.component',
  () => 'BookAppointmentLink',
)

it('<Dashboard/> render correctly', () => {
  const tree = renderer
    .create(
      <Dashboard
        origin={undefined}
        dispatch={() => {}}
        incompleteSurveys={0}
        unreadMessages={0}
        futureAppointmentsCount={0}
        clinics={[]}
        patient={{}}
        numberOfPatients={1}
        primaryConversation={'123'}
        onlyOneConversation={false}
        forceConversationId={0}
        primaryConversationId={0}
        scheduleAppointment={0}
        patients={[]}
      />,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
