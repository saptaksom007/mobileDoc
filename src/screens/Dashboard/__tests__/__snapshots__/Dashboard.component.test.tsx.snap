// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard/> render correctly 1`] = `
Array [
  <RCTScrollView
    accessibilityLabel="dashboard"
    contentContainerStyle={
      Object {
        "flex": 1,
        "justifyContent": "space-evenly",
        "marginBottom": 50,
      }
    }
    testID="dashboard"
  >
    <View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 37.5,
          }
        }
      >
        <Box
          badgeNumber={0}
          color="#8950b7"
          name="event"
          nextRoute={[Function]}
          testID="dashboard-nav-appointment"
        />
        <Box
          badgeNumber={0}
          color="#1a7dae"
          locked={true}
          name="question-answer"
          nextRoute={[Function]}
          testID="dashboard-nav-conversation"
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 37.5,
          }
        }
      >
        <Box
          color="#fba81f"
          locked={true}
          name="stacked-line-chart"
          nextRoute={[Function]}
          testID="dashboard-nav-survey"
        />
        <Box
          badgeNumber={0}
          color="#27debe"
          locked={true}
          name="description"
          nextRoute={[Function]}
          testID="dashboard-nav-dossier"
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 37.5,
          }
        }
      >
        <Box
          color="#2a6df7"
          locked={true}
          name="escalator-warning"
          nextRoute={[Function]}
          testID="dashboard-nav-dossier"
        />
        <Box
          color="#f37a9e"
          locked={true}
          name="officehours"
          nextRoute={[Function]}
          testID="dashboard-nav-availability"
        />
      </View>
    </View>
  </RCTScrollView>,
  <View
    pointerEvents="box-none"
    style={
      Object {
        "backgroundColor": "transparent",
        "bottom": 0,
        "elevation": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 0,
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "alignItems": "flex-end",
          "bottom": 94,
          "opacity": 0,
          "position": "absolute",
          "right": -1000,
          "zIndex": 10,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "elevation": 0,
            "flex": 1,
            "flexDirection": "column",
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "elevation": 0,
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 5,
              "paddingBottom": 8,
              "paddingLeft": 0,
              "paddingRight": 38,
              "paddingTop": 8,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#feffff",
                  "marginRight": 4,
                }
              }
            >
              appointmentslist.subtitle
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "elevation": 0,
            "flex": 1,
            "flexDirection": "column",
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "elevation": 0,
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 5,
              "paddingBottom": 8,
              "paddingLeft": 0,
              "paddingRight": 38,
              "paddingTop": 8,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#feffff",
                  "marginRight": 4,
                }
              }
            >
              exercise
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "elevation": 0,
            "flex": 1,
            "flexDirection": "column",
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "elevation": 0,
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 5,
              "paddingBottom": 8,
              "paddingLeft": 0,
              "paddingRight": 38,
              "paddingTop": 8,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#feffff",
                  "marginRight": 4,
                }
              }
            >
              nutrition
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "elevation": 0,
            "flex": 1,
            "flexDirection": "column",
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "elevation": 0,
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 5,
              "paddingBottom": 8,
              "paddingLeft": 0,
              "paddingRight": 38,
              "paddingTop": 8,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#feffff",
                  "marginRight": 4,
                }
              }
            >
              activity
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityLabel="Floating Action Button"
      accessible={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#4C748c",
          "borderRadius": 28,
          "bottom": 30,
          "elevation": 5,
          "height": 56,
          "justifyContent": "center",
          "opacity": 1,
          "overflow": "visible",
          "position": "absolute",
          "right": 30,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 5,
            "width": 0,
          },
          "shadowOpacity": 0.35,
          "shadowRadius": 3,
          "transform": Array [
            Object {
              "rotate": "0deg",
            },
            Object {
              "scale": 1,
            },
          ],
          "width": 56,
          "zIndex": 2,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 28,
            "height": 56,
            "justifyContent": "center",
            "opacity": 1,
            "width": 56,
            "zIndex": 3,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 28,
              "flex": 1,
              "height": 56,
              "justifyContent": "center",
              "transform": Array [
                Object {
                  "rotate": "0deg",
                },
              ],
              "width": 56,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "position": "absolute",
                    "width": 2,
                  },
                  Object {
                    "backgroundColor": "#fff",
                    "height": 15,
                  },
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "height": 2,
                    "position": "absolute",
                  },
                  Object {
                    "backgroundColor": "#fff",
                    "width": 15,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>,
]
`;
